evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+import pandas as pd
+from skopt.space import Integer
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+from evalml.utils import deprecate_arg, import_or_raise
+
+
+class ARIMARegressor(Estimator):
+    """
+    Autoregressive Integrated Moving Average Model.
+    The three parameters (p, d, q) are the AR order, the degree of differencing, and the MA order.
+    More information here: https://www.statsmodels.org/devel/generated/statsmodels.tsa.arima_model.ARIMA.html
+
+    """
+    name = "ARIMA Regressor"
+    hyperparameter_ranges = {
+        "p": Integer(0, 10),
+        "d": Integer(0, 10),
+        "q": Integer(0, 10),
+    }
+    model_family = ModelFamily.ARIMA
+    supported_problem_types = [ProblemTypes.TIME_SERIES_REGRESSION]
+
+    def __init__(self, date_column=None, p=1, d=0, q=0,
+                 random_seed=0, random_state=None, **kwargs):
+        random_seed = deprecate_arg("random_state", "random_seed", random_state, random_seed)
+        order = (p, d, q)
+        parameters = {'order': order}
+        parameters.update(kwargs)
+        self.date_column = date_column
+
+        p_error_msg = "ARIMA is not installed. Please install using `pip install statsmodels`."
+        arima = import_or_raise("statsmodels.tsa.arima_model", error_msg=p_error_msg)
+        arima.ARIMA(endog=np.zeros(p + d + q + 1), **parameters)  # to validate parameters
+        super().__init__(parameters=parameters,
+                         component_obj=None,
+                         random_seed=random_seed)
+
+    def get_dates_fit(self, X, y):
```

Comment:
`get_dates_fit` is doing two things: getting the dates from the column or index and then changing the index of `X` to match that of `y`. Can we you break up those two things into different functions? I think it'll be easier to follow the logic at that point. Also, once we're there I think we'll be able to reuse the method for getting the dates in `predict` and get rid of `get_dates_predict`?